perma_web/functional_tests/tests.py

```diff
@@ -134,22 +132,24 @@ class FunctionalTest(BaseTestCase):
     virtual_display = None
 
     def setUp(self):
-        if REMOTE_SERVER_URL:
-            self.server_url = REMOTE_SERVER_URL
-        else:
-            # By default, the test server only mounts the django app,
-            # which will leave out the warc app, so mount them both here
-            self.server_thread.httpd.set_app(self.server_thread.static_handler(wsgi_app))
-            self.server_thread.host = LOCAL_SERVER_DOMAIN
+            if REMOTE_SERVER_URL:
+                self.server_url = REMOTE_SERVER_URL
+            else:
+                # By default, the test server only mounts the django app,
+                # which will leave out the warc app, so mount them both here
+                self.server_thread.httpd.set_app(self.server_thread.static_handler(wsgi_app))
+                self.server_thread.host = LOCAL_SERVER_DOMAIN
 
-            self.server_url = self.live_server_url
+                self.server_url = self.live_server_url
 
-        if USE_SAUCE:
-            self.setUpSauce()
-        else:
-            self.setUpLocal()
+            if USE_SAUCE:
```

Comment:
Suggestion: this if/else block is the part that's too slow to repeat for each test, because that's where we spin up the headless browser. Maybe move this block to `setUpClass`, and just leave stuff that has to be there in `setUp`? Note that `tearDown` then has to become `tearDownClass` as well, because it's the inverse of this block.

Also, nitpick: we follow [pep8](https://www.python.org/dev/peps/pep-0008/) for python code, including 4 spaces for indent. This block became 8 spaces somehow. :)